src/nodes/flow/coupling_flows/planar_flow.jl

```diff
@@ -0,0 +1,258 @@
+getb(f::PlanarFlow)                         = return f.b
+getall(f::PlanarFlow)                       = return f.u, f.w, f.b
+getdim(f::PlanarFlowEmpty{N}) where { N }   = return N
+
+# set-functions for the PlanarFlow structure
+function setu!(f::PlanarFlow{T1,T2}, u::T1) where { T1, T2 <: Real}
+    @assert length(f.u) == length(u) "The dimensionality of the current value of u and its new value do not match."
+    f.u = u
+end
+
+function setw!(f::PlanarFlow{T1,T2}, w::T1) where { T1, T2 <: Real}
+    @assert length(f.w) == length(w) "The dimensionality of the current value of w and its new value do not match."
+    f.w = w
+end
+
+function setb!(f::PlanarFlow{T1,T2}, b::T2) where { T1, T2 <: Real }
+    f.b = b
+end
+
+# custom Base function for the PlanarFlow structure
+eltype(f::PlanarFlow{T1,T2}) where { T1 <: Real, T2 <: Real}                 = promote_type(T1, T2)
+eltype(f::PlanarFlow{T1,T2}) where { T1 <: AbstractArray, T2 <: Real}        = promote_type(eltype(T1), T2)
+eltype(::Type{PlanarFlow{T1,T2}}) where { T1 <: Real, T2 <: Real}            = promote_type(T1, T2)
+eltype(::Type{PlanarFlow{T1,T2}}) where { T1 <: AbstractArray, T2 <: Real}   = promote_type(eltype(T1), T2)
+
+size(f::PlanarFlow{T1,T2}) where { T1 <: Real, T2 <: Real}                   = 1
+size(f::PlanarFlow{T1,T2}) where { T1 <: AbstractArray, T2 <: Real}          = length(f.u)
+size(f::PlanarFlowEmpty{N}) where { N }                                      = return N
+
+length(f::PlanarFlow{T1,T2}) where { T1 <: Real, T2 <: Real}                 = 1
+length(f::PlanarFlow{T1,T2}) where { T1 <: AbstractArray, T2 <: Real}        = length(f.u)
+length(f::PlanarFlowEmpty{N}) where { N }                                    = return N
+
+# forward pass through the PlanarFlow function (multivariate input)
+function _forward(f::PlanarFlow{T1,T2}, input::T1) where { T1, T2 <: Real }
+
+    # fetch values
+    u, w, b = getall(f)
+    
+    # calculate result
+    result = zeros(size(u))
+    result .*= tanh(dot(w, input) + b) 
+    result .+= input
+
+    # return result
+    return result
+
+end
+forward(f::PlanarFlow{T1,T2}, input::T1) where { T1, T2 <: Real } = _forward(f, input)
+Broadcast.broadcasted(::typeof(forward), f::PlanarFlow{T1,T2}, input::Array{T1,1}) where { T1, T2 <: Real } = broadcast(_forward, Ref(f), input)
+
+
+# forward pass through the PlanarFlow function (univariate input)
+function _forward(f::PlanarFlow{T1,T2}, input::T3) where { T1 <: Real, T2 <: Real, T3 <: Real }
+
+    # fetch values
+    u, w, b = getall(f)
+    
+    # calculate result
+    result = copy(u)
+    result *= tanh(dot(w, input) + b)
+    result += input
+
+    # return result
+    return result
+
+end
+forward(f::PlanarFlow{T1,T2}, input::T3) where { T1 <: Real, T2 <: Real, T3 <: Real } = _forward(f, input)
+Broadcast.broadcasted(::typeof(forward), f::PlanarFlow{T1,T2}, input::Array{T3,1}) where { T1 <: Real, T2 <: Real, T3 <: Real } = broadcast(_forward, Ref(f), input)
```

Comment:
I would change concrete types everywhere, e.g. from `Array{T3, 1}` to `AbstractArray{T3, 1}` (or even `AbstractVector{T3}` but its minor, `AbstractArray` is also fine)

Reply:
Resolved by 0056d3a.